### PR TITLE
Added widening multiply for u16x8, u32x4, u32x8, i32x4 and i32x8

### PR DESCRIPTION
**Pull request by mcroomp**
Add implementations for all the intrinsics @Melirius might be interested

Also ran cargo fmt +nightly, so some comments got wrapped

### CODE REVIEW

File: src/i32x4_.rs [hunk 1]
```diff
@@ -490,6 +490,45 @@ impl i32x4 {
       }
     }
   }
+
+  #[inline]
+  #[must_use]
+  pub fn mul_widen(self, rhs: Self) -> i64x4 {
```
Comment (Lokathor) [inline src/i32x4_.rs:496]: could we put a doc comment here about what happens, like the others have.

**Review by Lokathor — CHANGES_REQUESTED** [review]

File: src/i32x4_.rs (hunk 1 above)
Comment (mcroomp) [reply src/i32x4_.rs:501]: Hi there, let me know if there is anything else in the PR. Thanks!

**Review by Lokathor — APPROVED** [review]

**Comment by Lokathor** [comment]
> Published wide v0.7.29 at registry `crates-io`

**Comment by mcroomp** [comment]
Don't forget to merge :)

**Comment by Lokathor** [comment]
i didn't merge it!! :0

**Comment by Lokathor** [comment]
okay now `0.7.30` should have it

**Pull request merged**